azkaban-common/src/main/java/azkaban/storage/LocalStorage.java

```diff
@@ -36,34 +37,32 @@
 public class LocalStorage implements Storage {
   private static final Logger log = Logger.getLogger(LocalStorage.class);
 
-  final File baseDirectory;
+  final File rootDirectory;
 
-  public LocalStorage(File baseDirectory) {
-    this.baseDirectory = validateBaseDirectory(createIfDoesNotExist(baseDirectory));
+  @Inject
+  public LocalStorage(AzkabanCommonModuleConfig config) {
```

Comment:
I think classes operating on a config is a standard convention. Just that our module is too big - as in common has too much stuff in it. But for the time being I think, it is easier to work with a common config container since it also contains props as well.

So my suggestion is to use this model if it applies - for example if you need a bunch of `props` values and need to extract information from them.